puzzles/views.py

```diff
@@ -128,19 +129,35 @@ def add_tag(request, pk):
                 name=form.cleaned_data["name"],
                 defaults={'color' : form.cleaned_data["color"]}
             )
-            puzzle.tags.add(tag)
+            if tag.is_meta:
+                # the post m2m hook will add tag
+                puzzle.metas.add(Puzzle.objects.get(name=tag.name))
```

Comment:
What if puzzles.metas already includes this meta? (is this even possible)

Reply:
It's possible that someone adds a tag at the same time that another person adds a puzzle to a meta, but either way one puzzle.metas.add will go through and trigger the m2m hook to sync the tags.